services/datamanager/data_manager.go

```diff
@@ -89,8 +108,7 @@ type DataCaptureConfig struct {
 
 // Equals checks if one capture config is equal to another.
 func (c *DataCaptureConfig) Equals(other *DataCaptureConfig) bool {
-	return c.Resource == other.Resource &&
```

Comment:
Removing this line ultimately fixed the bug.  There is no way to specify the `Resource` field in the JSON itself, so we should not be doing an equality check on this field in the resource.  The resource is assigned to this struct in the `updateDataCaptureConfigs` which is not called until reconfiguration (which only happens after the deep equals on the config is called).

IMO its a little hacky to be mutating the config struct this way and it would make more sense to refactor it so that it only contains fields specified in JSON rather than needing to be added to in order for it to be a valid struct.  But I won't do that in this PR unless people request it.